src/lava/proc/graded/process.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import os
+import numpy as np
+import typing as ty
+from typing import Any, Dict
+from enum import IntEnum, unique
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.variable import Var
+from lava.magma.core.process.ports.ports import InPort, OutPort
+
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+
+
+def loihi2round(vv):
+    """
+    Round values in numpy array the way loihi 2
+    performs rounding/truncation.
+    """
+    return np.fix(vv + (vv > 0) - 0.5).astype('int')
+
+
+class GradedVec(AbstractProcess):
+    """GradedVec
+    Thresholded graded spike vector
+
+    Parameters
+    ----------
+    shape: tuple(int)
+        number and topology of neurons
+    vth: int
+        threshold for spiking
+    exp: int
+        fixed point base
+    """
+
+    def __init__(
+            self,
+            *,
+            shape: ty.Tuple[int, ...],
+            vth=1,
+            exp=0) -> None:
+
+        super().__init__(shape=shape)
+
+        self.a_in = InPort(shape=shape)
+        self.s_out = OutPort(shape=shape)
+
+        self.v = Var(shape=shape, init=0)
+        self.vth = Var(shape=(1,), init=vth)
+        self.exp = Var(shape=(1,), init=exp)
+
+    @property
+    def shape(self) -> ty.Tuple[int, ...]:
+        """Return shape of the Process."""
+        return self.proc_params['shape']
+
+
+class NormVecDelay(AbstractProcess):
+    """NormVec
+    Normalizable graded spike vector
```

Comment:
All vectors are normalizable, but this one is normalized (or even, self-normalizing).

Reply:
well not vectors on loihi 2. this isnt self normalizing either as it just one of the procs needed for the self-normalizing network.